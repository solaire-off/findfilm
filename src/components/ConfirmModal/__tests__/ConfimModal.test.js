import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { ConfirmModal } from "..";

it("Correct render", () => {
  const { asFragment } = render(
    <ConfirmModal
      confirmCallback={() => {}}
      modalTitle="Confirm title"
      modalDescr="Confirm descr"
    />
  );

  expect(asFragment()).toMatchSnapshot();
});

it("Work confim callback", () => {
  const mockCallback = jest.fn();
  const mockClose = jest.fn();

  render(
    <ConfirmModal
      confirmCallback={mockCallback}
      closeModal={mockClose}
      modalTitle="Confirm title"
      modalDescr="Confirm descr"
    />
  );

  userEvent.click(screen.getByText(/Confirm/i, { selector: "button" }));

  expect(mockCallback).toBeCalledTimes(1);
  expect(mockClose).toBeCalledTimes(1);
});
