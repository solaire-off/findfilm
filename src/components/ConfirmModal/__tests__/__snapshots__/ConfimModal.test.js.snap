// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Correct render 1`] = `
<DocumentFragment>
  <div
    class="modal__header"
  >
    <p
      class="modal__title"
    >
      Confirm title
    </p>
    <button
      aria-label="Close"
      class="modal__close"
      type="button"
    />
  </div>
  <div
    class="modal__body"
  >
    <div
      class="modal__text"
    >
      <p>
        Confirm descr
      </p>
    </div>
  </div>
  <div
    class="modal__footer modal__footer--mt-sm"
  >
    <button
      class="btn btn--primary btn--lg modal__btn font-weight-medium"
      type="button"
    >
      Confirm
    </button>
  </div>
</DocumentFragment>
`;
